sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -309,6 +309,71 @@ static JsonNode parseResponse(HttpEntity responseEntity) throws IOException {
     return responses;
   }
 
+  @AutoValue
+  public abstract static class SslConfiguration implements Serializable {
+    public abstract @Nullable String getKeystoreType();
+
+    public abstract @Nullable String getKeystorePath();
+
+    public abstract @Nullable String getKeystorePassword();
+
+    public abstract @Nullable TrustStrategy getTrustStrategy();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setKeystoreType(String type);
```

Comment:
Aside from reorganizing classes, the biggest win of this PR is exposing the ability for pipeline developers to use their keystore of choice- jks, pkcs12, etc